components/brave_wallet/browser/brave_wallet_provider_impl.cc

```diff
@@ -608,7 +665,14 @@ void BraveWalletProviderImpl::Locked() {
 }
 
 void BraveWalletProviderImpl::Unlocked() {
-  UpdateKnownAccounts();
+  if (pending_request_ethereum_permissions_callback_) {
+    RequestEthereumPermissions(
+        std::move(pending_request_ethereum_permissions_callback_));
+    pending_request_ethereum_permissions_callback_ =
+        RequestEthereumPermissionsCallback();
```

Comment:
nit: I'll prefer this to be taken out if it turns out to be unnecessary.

Reply:
Removed in rebase